File: back/src/main/java/com/openclassrooms/mddapi/models/Article.java
package com.openclassrooms.mddapi.models;

import java.util.Date;

import javax.persistence.*;

import lombok.Data;

@Data
@Entity
@Table(name = "articles")
public class Article {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  private String title;

  @Column(name = "user_username")
  private String userUsername;

  @Column(name = "theme_id")
  private Integer themeId;

  private String content;

  @Column(name = "created_at")
  private Date createdAt;

  @PrePersist
  protected void onCreate() {
    createdAt = new Date();
  };
};

File: front/src/app/core/models/Article.ts
export interface Article {
  id: number;
  title: string;
  userUsername: string;
  themeId: number;
  content: string;
  createdAt: string;
}

export type ArticleCreate = Pick<Article, 'userUsername' | 'themeId' | 'title' | 'content'>;